File: output/Nightwatch/SimpleStore/Fable.ReactNative.SimpleStore.KeyValueStore.js
import async$002Dstorage from "@react-native-community/async-storage";

const asyncStorage = async$002Dstorage;

export function getAllKeys() {
    return new Promise(((success, fail) => {
        const value = asyncStorage.getAllKeys(((err, keys) => {
            if ((!(err == null)) ? (!(err.message == null)) : false) {
                fail(err);
            }
            else {
                success(keys);
            }
        }));
        void value;
    }));
}

File: src/Client/SimpleStore/Fable.ReactNative.SimpleStore.KeyValueStore.fs
module Fable.ReactNativeSimpleStore.KeyValueStore
open Fable.Core.JsInterop

[<AutoOpen>]
module Helpers =
    let private asyncStorage : obj = importDefault "@react-native-community/async-storage"

    /// Retrieves all keys from the AsyncStorage.
    let getAllKeys() : Fable.Core.JS.Promise<string []> =
        Promise.create(fun success fail ->
            asyncStorage?getAllKeys
                (fun err keys ->
                    if not (isNull err) && not (isNull (err?message)) then
                        fail (unbox err)
                    else
                        success (unbox keys)) |> ignore)